src/addiction.cpp

```diff
@@ -1,57 +1,61 @@
 #include "addiction.h"
+#include "debug.h"
+#include "pldata.h"
+#include "player.h"
+#include "morale.h"
 #include "rng.h"
-#include "game.h"
 #include "messages.h"
 
-void addict_effect(addiction &add)
+void addict_effect(player &u, addiction &add,
+                   std::function<void (char const*)> const &cancel_activity)
 {
-    int in = add.intensity;
+    int const in = add.intensity;
 
     switch (add.type) {
     case ADD_CIG:
         if (in > 20 || one_in((500 - 20 * in))) {
             add_msg(rng(0, 6) < in ? _("You need some nicotine.") :
                     _("You could use some nicotine."));
```

Comment:
This (and others) could probably be `u.add_msg_if_player(...)`, to forestall the likely far-future issue of the player getting messages for an NPC's nicotine addiction (I'm not sure that's possible at present, but it really should be).

Technically, doing so might even let you get away with removing the include of `messages.h`, but that's unlikely to make much difference.